.github/workflows/upgrade-submodules.yml

```diff
@@ -0,0 +1,39 @@
+name: upgrade-submodules
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * 5' # 9:00AM on Friday
+jobs:
+  open_pull_request:
+    if: github.repository == 'saleor/saleor-platform'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout recursively
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Configure git
+        run: |
+          git config user.name 'DevRel Bot'
+          git config user.email 'noreply@saleor.io'
+      - name: Update submodules
+        run: |
+          git submodule foreach git fetch origin
+          git submodule foreach git checkout main
```

Comment:
We are fetching main branches instead of the latest tag. Wonder if it will introduce issues with incompatible versions. A manual smoke test will be needed before the merge.

Minor thing - not sure if that's still the case, but was in the past - main branches do not report the saleor version in the dashboard.  (if you go to the dashboard -> configuration there's a version of all the modules). It might be a bit confusing for the users to not be able to easily check version they are using.

Reply:
agreed, this should be tag based.

Reply:
> We are fetching main branches instead of the latest tag. Wonder if it will introduce issues with incompatible versions. A manual smoke test will be needed before the merge

Here are multiple topics to be taken into consideration:
 * what version user would anticipate, when fetching that repository for the first time? I would assume that this will be tip of development branch (therefore `main`), ultimately we try to ensure that both dashboard and core work flawlessly with each other on that branch,
 * wo do not release storefront (yet), so what you suggest is basically: core and dasbhoard from release branches, storefront from `main` (we have a discussion about releasing storefront at the back of our heads with @lkostrowski )
 * ultimately I would recommend - if we want to provide release-based platform, to maintain these releases on their dedicated branches (so if you want to check how 3.5 would play, just checkout here into `3.5` branch)

> Minor thing - not sure if that's still the case, but was in the past - main branches do not report the saleor version in the dashboard. (if you go to the dashboard -> configuration there's a version of all the modules). It might be a bit confusing for the users to not be able to easily check version they are using.

AFAIK it's currently reported (I would not consider that _corect_, but that's the same value we show cloud users when they create environment from `main` branch, currently it's `dashboard v3.6.0-dev
core v3.6.0-a`)

Reply:
I think about something sophisticated, like tags (which seems more suitable than branches here), that are specific to work together. Eg. using semver. In general, I'd be safer if we use tags than `main` - can we be always sure `main` works?

But this PR is a good MVP. It's nondestructive, so we can always check PR. 
The first milestone is to make sure we update this repo - having notification that PR was opened is good to me (PR can be linked to the project).

So approved at this stage from me, and we can continue the discussion about the future.

Reply:
> what version user would anticipate, when fetching that repository for the first time? I would assume that this will be tip of development branch (therefore main), ultimately we try to ensure that both dashboard and core work flawlessly with each other on that branch,

Usually, when I pull docker images, I assume it's gonna be the latest stable, not the dev branch. Don't think it would cause issues most of the time, but could be an issue when API has introduced breaking changes, and the dashboard did not update yet. 
It can be addressed in the next version of the update script, and for now we'll check it manually.

> ultimately I would recommend - if we want to provide release-based platform, to maintain these releases on their dedicated branches (so if you want to check how 3.5 would play, just checkout here into 3.5 branch)

Yup, that would be awesome, but more scripts would be involved to do that

>AFAIK it's currently reported (I would not consider that corect, but that's the same value we show cloud users when they create environment from main branch, currently it's dashboard v3.6.0-dev core v3.6.0-a)

So it changed since my last try. Thank you for checking!